tests/src/test/java/alluxio/server/ft/journal/JournalBackupIntegrationTest.java

```diff
@@ -129,6 +143,99 @@ public void backupRestoreMetastore_Rocks() throws Exception {
+    emergencyBackupCore(1);
+  }
+
+  @Test
+  public void emergencyBackupHA() throws Exception {
+    emergencyBackupCore(3);
+  }
+
+  private void emergencyBackupCore(int numMasters) throws Exception {
+    TemporaryFolder backupFolder = new TemporaryFolder();
+    backupFolder.create();
+    List<PortCoordination.ReservedPort> ports1 = numMasters > 1
+        ? PortCoordination.BACKUP_EMERGENCY_HA_1 : PortCoordination.BACKUP_EMERGENCY_1;
+    String clusterName1 = numMasters > 1 ? "emergencyBackup_HA_1" : "emergencyBackup_1";
+    mCluster = MultiProcessCluster.newBuilder(ports1)
+        .setClusterName(clusterName1)
+        .setNumMasters(numMasters)
+        .setNumWorkers(0)
+        // Masters become primary faster, will be ignored in non HA case
+        .addProperty(PropertyKey.ZOOKEEPER_SESSION_TIMEOUT, "1sec")
+        .addProperty(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.UFS)
+        .addProperty(PropertyKey.MASTER_METASTORE, MetastoreType.ROCKS)
+        .addProperty(PropertyKey.MASTER_BACKUP_DIRECTORY, backupFolder.getRoot())
+        .addProperty(PropertyKey.MASTER_JOURNAL_BACKUP_WHEN_CORRUPTED, true)
+        .build();
+    mCluster.start();
+    final int numFiles = 10;
+    // create normal uncorrupted journal
+    for (int i = 0; i < numFiles; i++) {
+      mCluster.getFileSystemClient().createFile(new AlluxioURI("/normal-file-" + i));
+    }
+    mCluster.stopMasters();
+    // corrupt journal
+    URI journalLocation = new URI(mCluster.getJournalDir());
+    UfsJournal fsMaster =
+        new UfsJournal(URIUtils.appendPathOrDie(journalLocation, "FileSystemMaster"),
+            new NoopMaster(), 0, Collections::emptySet);
+    fsMaster.start();
+    fsMaster.gainPrimacy();
+    long nextSN = 0;
+    try (UfsJournalReader reader = new UfsJournalReader(fsMaster, true)) {
+      while (reader.advance() != JournalReader.State.DONE) {
+        nextSN++;
+      }
+    }
+    try (UfsJournalLogWriter writer = new UfsJournalLogWriter(fsMaster, nextSN)) {
+      Journal.JournalEntry entry = Journal.JournalEntry.newBuilder()
+          .setSequenceNumber(nextSN)
+          .setDeleteFile(alluxio.proto.journal.File.DeleteFileEntry.newBuilder()
+              .setId(4563728) // random non-zero ID number (zero would delete the root)
+              .setPath("/nonexistant")
+              .build())
+          .build();
+      writer.write(entry);
+      writer.flush();
+    }
+    // this should fail and create a backup
+    mCluster.startMasters();
+    // assert and find backup file
+    CommonUtils.waitFor("backup file to be created automatically",
+        () -> 2 == Objects.requireNonNull(backupFolder.getRoot().list()).length,
+        WaitForOptions.defaults().setInterval(500).setTimeoutMs(30_000));
+    Optional<String> backupFile =
+        Arrays.stream(Objects.requireNonNull(backupFolder.getRoot().list()))
+            .filter(s -> s.endsWith(".gz")).findFirst();
+    assertTrue(backupFile.isPresent());
```

Comment:
In HA these files should have the same contents right? Maybe check each file has the same entires?

Reply:
I expanded the integration test so it would test every backup in HA mode, not just a single one.